packages/tools/devtools/devtools-view/src/components/TelemetryView.tsx

```diff
@@ -89,6 +90,12 @@ export function TelemetryView(): React.ReactElement {
 				setTelemetryEvents(message.data.contents);
 				return true;
 			},
+			[CloseDevtools.MessageType]: (untypedMessage) => {
```

Comment:
While this will fix the issue for telemetry specifically, I think we can tackle this whole class of problem by moving this check into `DevtoolsView` and reset all local state when we receive the message. That will result in the telemetry view being de-selected and its content being cleared.

Reply:
Updated.